app/controllers/v0/message_drafts_controller.rb

```diff
@@ -4,36 +4,40 @@ class MessageDraftsController < SMController
     def create
       draft = MessageDraft.new(draft_params)
       raise Common::Exceptions::ValidationErrors, draft unless draft.valid?
-      response = client.post_create_message_draft(draft_params)
 
-      render json: response,
+      draft_response = client.post_create_message_draft(draft_params)
+      render json: draft_response,
              serializer: MessageSerializer,
              status: :created
     end
 
     def update
-      draft = MessageDraft.new(draft_params)
-      raise Common::Exceptions::ValidationErrors, draft unless draft.valid?
+      draft = client.post_create_message_draft(draft_params.merge(id: params[:id]))
 
-      client.post_create_message_draft(draft_params.merge(id: params[:id]))
+      # draft will fail validation if its a reply draft.
+      raise Common::Exceptions::ValidationErrors, draft unless draft.valid?
       head :no_content
     end
 
     def create_reply_draft
-      draft = MessageDraft.new(reply_draft_params).as_reply
+      draft = client.post_create_message_draft_reply(params[:reply_id], reply_draft_params)
+
+      # draft will fail validation if its not a reply draft.
       raise Common::Exceptions::ValidationErrors, draft unless draft.valid?
```

Comment:
I think the reason I'm confused on this is that this should have already failed validation of some sort before the call to client.post_create... aren't we in affact changing something and then raising validation error after the fact?